src/lib/proof-system/zkprogram.ts

```diff
@@ -656,22 +664,33 @@ function ZkProgram<
   async function compile({
     cache = Cache.FileSystemDefault,
     forceRecompile = false,
+    proofsEnabled = true,
   } = {}) {
+    doProving = proofsEnabled;
+
     let methodsMeta = await analyzeMethods();
     let gates = methodKeys.map((k) => methodsMeta[k].gates);
```

Comment:
can't we skip this fairly expensive step when proofs are disabled?

Reply:
we can, good point!